agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -2580,6 +2608,47 @@ bool backhaul_manager::handle_slave_ap_metrics_response(ieee1905_1::CmduMessageR
         std::copy_n(response.metric.estimated_service_info_field.begin(),
                     response.metric.estimated_service_info_field.size(),
                     ap_metrics_response_tlv->estimated_service_info_field());
+
+        if (!response.sta_traffic_stats.empty()) {
+            for (auto stat : response.sta_traffic_stats) {
+
+                auto sta_traffic_response_tlv =
+                    cmdu_tx.addClass<wfa_map::tlvAssociatedStaTrafficStats>();
+
+                if (!sta_traffic_response_tlv) {
+                    LOG(ERROR) << "Failed addClass<wfa_map::tlvAssociatedStaTrafficStats>";
+                }
+
+                sta_traffic_response_tlv->sta_mac()              = stat.sta_mac;
+                sta_traffic_response_tlv->byte_sent()            = stat.byte_sent;
+                sta_traffic_response_tlv->byte_recived()         = stat.byte_recived;
+                sta_traffic_response_tlv->packets_sent()         = stat.packets_sent;
+                sta_traffic_response_tlv->packets_recived()      = stat.packets_recived;
+                sta_traffic_response_tlv->tx_packets_error()     = stat.tx_packets_error;
+                sta_traffic_response_tlv->rx_packets_error()     = stat.rx_packets_error;
+                sta_traffic_response_tlv->retransmission_count() = stat.retransmission_count;
+            }
+        }
+
+        if (!response.sta_link_metrics.empty()) {
+            for (auto link_metric : response.sta_link_metrics) {
+
+                auto sta_link_metric_response_tlv =
+                    cmdu_tx.addClass<wfa_map::tlvAssociatedStaLinkMetrics>();
+
+                if (!sta_link_metric_response_tlv) {
+                    LOG(ERROR) << "Failed addClass<wfa_map::tlvAssociatedStaLinkMetrics>";
+                }
+
+                sta_link_metric_response_tlv->sta_mac() = link_metric.sta_mac;
+                if (!sta_link_metric_response_tlv->alloc_bssid_info_list(1)) {
```

Comment:
Maybe you should copy as many entries as there are in the source TLV to the destination TLV (instead of supposing that there is just one) as it is being done in some other parts of the code where a `bssid_info_list` is copied.

Reply:
No, we have just 1 bssid_info_list.